packages/pn-commons/src/components/Header/Header.tsx

```diff
@@ -58,65 +53,42 @@ const Header = ({
   onAssistanceClick,
   eventTrackingCallbackProductSwitch,
   isLogged,
-  selfcareBaseUrl,
-  selfcareSendProdId,
 }: HeaderProps) => {
   const pagoPAHeaderLink: RootLinkType = {
     ...pagoPALink(),
     label: 'PagoPA S.p.A.',
     title: getLocalizedOrDefaultLabel('common', 'header.pago-pa-link', 'Sito di PagoPA S.p.A.'),
   };
 
-  const [isHeaderReady, setIsHeaderReady] = useState(false);
-
-  const trackEvent = (url: string) => {
-    if (eventTrackingCallbackProductSwitch) {
-      eventTrackingCallbackProductSwitch(url);
-    }
-  };
-
   const handleProductSelection = (product: ProductEntity) => {
-    if (!partyId) {
-      return;
+    if (eventTrackingCallbackProductSwitch) {
+      eventTrackingCallbackProductSwitch(product.productUrl);
     }
-    // eslint-disable-next-line functional/no-let
-    let url: string;
-    /** Here is necessary to clear sessionStorage otherwise when navigating throgh Area Riservata
-     * We enter in a state when the user was previously logged in but Area Riservata and PN require
-     * another token-exchange. Since the user "seems" to be logged in due to the presence of the old token
-     * in session storage, the token-exchange doesn't happen and the application enters in a blank state
-     * Further analysis is required when the navigation from PN will be allowed also to other products and
-     * not only to Area Reservata
-     *
-     * Carlotta Dimatteo 07/11/2022
-     */
-    sessionStorage.clear();
-
-    if (product.id === 'selfcare') {
-      url = `${selfcareBaseUrl}/dashboard/${partyId}`;
-      window.location.assign(url);
-      trackEvent(url);
-      return;
+    if (product.productUrl) {
+      /* eslint-disable-next-line functional/immutable-data */
+      window.location.assign(product.productUrl);
+      /** Here is necessary to clear sessionStorage otherwise when navigating throgh Area Riservata
+       * We enter in a state when the user was previously logged in but Area Riservata and PN require
+       * another token-exchange. Since the user "seems" to be logged in due to the presence of the old token
+       * in session storage, the token-exchange doesn't happen and the application enters in a blank state
+       * Further analysis is required when the navigation from PN will be allowed also to other products and
+       * not only to Area Reservata
+       *
+       * Carlotta Dimatteo 07/11/2022
+       */
+      sessionStorage.clear();
     }
-
-    url = `${selfcareBaseUrl}/token-exchange?institutionId=${partyId}&productId=${product.id}`;
-    window.location.assign(url);
-    trackEvent(url);
   };
 
-  const handlePartySelection = (party: PartyEntity) => {
-    window.location.assign(
-      `${selfcareBaseUrl}/token-exchange?institutionId=${party.id}&productId=${selfcareSendProdId}`
-    );
+  const handlePartySelection = (party: PartyEntityWithUrl) => {
+    window.location.assign(party.entityUrl);
```

Comment:
add the check that party.entityUrl is defined and add the test case.
So leave a mocked entity with empty url. Add test case where we click on entity with empty url (no assign called) and click on entity with filled url (assign called)